frontend-react/src/hooks/UseValueSets.ts

```diff
@@ -0,0 +1,152 @@
+
+export interface TableAttributes {
+    version: number;
+    createdAt?: string;
+    createdBy?: string;
+}
+
+/*
+
+  Fetch / Mutate functions and helpersa
+
+*/
+
+/*
+TO DISCUSS: we could set the pattern to be something like this or
+we could reference query keys within the function https://stackoverflow.com/a/68111112/5500298
+
+Have not tested, but I feel like to get the full value of react-query's refetching behavior we may want to go with the last option.
+When react-query refetches it will just re-run the function we passed to it, this would not take into account changes in variable dependencies
+unless we bake those into function itself?
+
+Note that I ran into type errors when trying to use higher order functions to avoid the inlining
+*/
+
+// will return either a value set or value set row depeding on
+// value set table name
+// we may not need the generic here, react-query probably handles that. But it couldn't hurt?
+const getLatestData = <T>(tableName: LookupTables, version: number) => {
+    const endpointHeader = lookupTableApi.getTableData<T>(version, tableName);
+
+    return axios
+        .get<T>(endpointHeader.url, endpointHeader)
+        .then((response) => response.data);
+};
+
+// not going to type this yet, as this function will likely go away when we bring in fetch by provider
+// returns a list of all lookup tables to be filtered
+const getLookupTables = () =>
+    axios(lookupTableApi.getTableList()).then(({ data }) => data);
+
+/*
+
+  Helper function to find the table we want within the response for ALL tables
+  Hopefully this will go away with the API refactor
+
+*/
+const findTableByName = (
+    tables: LookupTable[],
+    tableName: LookupTables
+): TableAttributes => {
+    const filteredBody: LookupTable[] = tables.filter(
+        (tv: LookupTable) => tv.tableName === tableName && tv.isActive
+    );
+
+    if (!filteredBody.length) {
+        throw new Error(`Table '${tableName}' was not found!`);
+    }
+    const table: LookupTable = filteredBody.sort(
+        (a: LookupTable, b: LookupTable) =>
+            b["tableVersion"] - a["tableVersion"]
+    )[0];
+
+    const { tableVersion, createdAt, createdBy } = table;
+    if (tableVersion === undefined) {
+        throw new Error(`No version of table '${tableName}' was found!`);
+    }
+
+    return {
+        version: tableVersion,
+        createdAt,
+        createdBy,
+    };
+};
+
+/*
+
+  useQuery function
+
+  hook used to get value sets and value set rows (defined by passsed dataTableName)
+
+*/
+
+export const useValueSetsTable = <T extends ValueSet | ValueSetRow>(
+    dataTableName: LookupTables,
+    suppliedVersion?: number
+): {
+    valueSetArray: Array<T>;
+    error: any;
+} => {
+    let versionData: Partial<TableAttributes> | null;
+    let error;
+    let valueSetArray;
+
+    // get all lookup tables
+    const { error: tableError, data: tableData } = useQuery<LookupTable[]>(
+        ["lookupTables"],
```

Comment:
I'm a little confused on what the key does. Are the integral enough that we should be enumerating them?

Reply:
down the line, definitely. I thought about that a bit this morning, and I think that this is something we can handle with our approach to resource classes. Without getting into the business of defining endpoints the way rest-hooks does, we can put some configuration into the resource class to keep track of the query keys for later reference.

as far as what they're used for, it's cache related stuff for the most part. One thing I mentioned in the comment [here](https://github.com/CDCgov/prime-reportstream/pull/6476/files#diff-82cdafd58a9ecaad3fbce3add39b460b71607726c282ed734647b7434a1386aaR25), though, is that you can also pass values in the keys for things like query params that, in addition to defining the cache storage location, also can be used within the fetch function. This is a decent resource on that, the official docs hint at it but don't really show it in action: 

https://stackoverflow.com/a/68111112/5500298

We may want to dig into that pattern later, but it is a little more complex and with what I've done so far it seems to work fine without it

Reply:
Ahh, yeah I was having trouble deciphering the pattern in that post but I se now...it uses the array with the key first, and trailing with params. Interesting that we have that option. I'm with you on it being more complex. Wonder what, if any, benefit that has or if it was built in strictly for having an option.